bolt-sidecar/src/builder/payload_builder.rs

```diff
@@ -243,6 +227,42 @@ impl FallbackPayloadBuilder {
             i += 1;
         }
     }
+
+    /// Fetch the previous RANDAO value from the beacon chain.
+    ///
+    /// NOTE: for some reason, using the ApiResult from `beacon_api_client` doesn't work, so
+    /// we are making a direct request to the beacon client endpoint.
+    async fn get_prev_randao(&self) -> Result<B256, BuilderError> {
+        let url = self
+            .beacon_api_client
+            .endpoint
+            .join("/eth/v1/beacon/states/head/randao")
+            .map_err(|e| BuilderError::Custom(format!("Failed to join URL: {e:?}")))?;
+
+        reqwest::Client::new()
+            .get(url)
+            .send()
+            .await?
+            .json::<Value>()
+            .await?
+            .pointer("/data/randao")
+            .and_then(|value| value.as_str())
+            .map(|value| B256::from_hex(value).map_err(BuilderError::Hex))
+            .ok_or_else(|| BuilderError::Custom("Failed to fetch prev_randao".to_string()))?
+    }
+
+    /// Fetch the expected withdrawals for the given slot from the beacon chain.
+    async fn get_expected_withdrawals_at_head(
+        &self,
+    ) -> Result<Vec<reth_primitives::Withdrawal>, BuilderError> {
```

Comment:
```suggestion
    ) -> Result<Vec<Withdrawal>, BuilderError> {
```
nit - better to import at the top? :)